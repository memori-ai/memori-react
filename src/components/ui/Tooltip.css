/* stylelint-disable no-descending-specificity */

.memori-tooltip {
    position: relative;
    display: inline-block;
}

.memori-tooltip--content {
    position: absolute;
    z-index: 100;
    top: 50%;
    left: 100%;
    display: none;
    width: auto;
    min-width: 200px;
    padding: 10px;
    border-radius: 10px;
    margin-left: 15px;
    background: rgba(0, 0, 0, 0.8);
    color: #fff;
    font-size: 0.85em;
    opacity: 0;
    text-align: center;
    transform: none;
    transition: 0.3s 0.1s all ease;
}

.memori-tooltip.memori-tooltip--align-left .memori-tooltip--content,
.memori-tooltip.memori-tooltip--align-topLeft .memori-tooltip--content {
    right: 100%;
    left: initial;
    margin: initial;
    margin-right: 15px;
}

.memori-tooltip .memori-tooltip--content::before {
    position: absolute;
    top: 50%;
    left: -15px;
    display: none;
    border: 10px solid rgba(0, 0, 0, 0.8);
    border-color: transparent rgba(0, 0, 0, 0.8) transparent transparent;
    margin-left: -5px;
    content: '';
    opacity: 0;
    transform: translateY(-50%);
    transition: 0.3s;
}

.memori-tooltip.memori-tooltip--align-left .memori-tooltip--content::before {
    right: -15px;
    left: auto;
    border-color: transparent transparent transparent rgba(0, 0, 0, 0.8);
    margin-right: -5px;
    margin-left: 0;
}

.memori-tooltip.memori-tooltip--align-topRight .memori-tooltip--content::before {
    top: 100%;
    right: auto;
    left: 10px;
    border-color: rgba(0, 0, 0, 0.8) transparent transparent transparent;
    margin-top: 10px;
    margin-left: 5px;
}

.memori-tooltip:not(.memori-tooltip--disabled).memori-tooltip--visible .memori-tooltip--content,
.memori-tooltip:not(.memori-tooltip--disabled).memori-tooltip--visible .memori-tooltip--content::before {
    display: block;
    opacity: 1;
}

.memori-tooltip:not(.memori-tooltip--disabled).memori-tooltip--visible .memori-tooltip--content,
.memori-tooltip:not(.memori-tooltip--disabled):not(.memori-tooltip--visible):hover .memori-tooltip--content {
    transform: translateY(-50%);
}

.memori-tooltip:not(.memori-tooltip--visible):not(.memori-tooltip--disabled):hover .memori-tooltip--content,
.memori-tooltip:not(.memori-tooltip--visible):not(.memori-tooltip--disabled):hover .memori-tooltip--content::before {
    display: block;
    opacity: 1;
}

.memori-tooltip.memori-tooltip--align-topRight:not(.memori-tooltip--disabled).memori-tooltip--visible
    .memori-tooltip--content,
.memori-tooltip.memori-tooltip--align-topRight:not(.memori-tooltip--disabled):not(.memori-tooltip--visible):hover
    .memori-tooltip--content {
    transform: translateY(-150%) translateX(-33%);
}

.memori-tooltip.memori-tooltip--align-topLeft .memori-tooltip--content::before {
    top: 100%;
    right: 10px;
    left: auto;
    border-color: rgba(0, 0, 0, 0.8) transparent transparent transparent;
    margin-top: 10px;
    margin-left: 5px;
}

.memori-tooltip.memori-tooltip--align-topLeft:not(.memori-tooltip--disabled).memori-tooltip--visible
    .memori-tooltip--content,
.memori-tooltip.memori-tooltip--align-topLeft:not(.memori-tooltip--disabled):not(.memori-tooltip--visible):hover
    .memori-tooltip--content {
    transform: translateY(-150%) translateX(33%);
}

.memori-tooltip:not(.memori-tooltip--disabled).memori-tooltip--visible .memori-tooltip--content a,
.memori-tooltip:not(.memori-tooltip--disabled):not(.memori-tooltip--visible):hover .memori-tooltip--content a {
    color: #fff;
    text-decoration: underline;
}

.memori-tooltip.memori-tooltip--disabled {
    cursor: not-allowed;
}
