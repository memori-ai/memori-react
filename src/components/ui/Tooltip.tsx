import React, { FC } from 'react';
import cx from 'classnames';

export interface Props {
  content: string | JSX.Element | React.ReactNode;
  className?: string;
  align?: 'left' | 'right' | 'topLeft' | 'topRight';
  disabled?: boolean;
  children: React.ReactNode;
  visible?: boolean;
}

const Tooltip: FC<Props> = ({
  content,
  className,
  align = 'right',
  disabled = false,
  visible = false,
  children,
}) => (
  <div
    className={cx(
      'memori-tooltip',
      `memori-tooltip--align-${align}`,
      className,
      {
        'memori-tooltip--disabled': disabled,
        'memori-tooltip--visible': visible,
      }
    )}
  >
    <div className="memori-tooltip--content">{content}</div>
    <div className="memori-tooltip--trigger">{children}</div>
  </div>
);

export default Tooltip;
