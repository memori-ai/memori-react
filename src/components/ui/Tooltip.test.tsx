import React from 'react';
import { render } from '@testing-library/react';
import Tooltip from './Tooltip';

it('renders Tooltip unchanged', () => {
  const { container } = render(
    <Tooltip content="Here is some tooltip content">Hover me</Tooltip>
  );
  expect(container).toMatchSnapshot();
});

it('renders Tooltip with jsx content unchanged', () => {
  const { container } = render(
    <Tooltip
      content={
        <p>
          Here we have some JSX, useful for
          <a href="https://memori.ai" rel="noopener noreferrer" target="_blank">
            links
          </a>
          , for example.
        </p>
      }
    >
      Hover me
    </Tooltip>
  );
  expect(container).toMatchSnapshot();
});

it('renders Tooltip disabled unchanged', () => {
  const { container } = render(
    <Tooltip content="Here is some tooltip content but I am disabled" disabled>
      Hover me
    </Tooltip>
  );
  expect(container).toMatchSnapshot();
});

it('renders Tooltip visible unchanged', () => {
  const { container } = render(
    <Tooltip
      content="Here is some tooltip content by default as I am controlled"
      visible
    >
      Hover me
    </Tooltip>
  );
  expect(container).toMatchSnapshot();
});

it('renders Tooltip aligned left unchanged', () => {
  const { container } = render(
    <Tooltip
      content="Here is some tooltip content by default as I am controlled"
      visible
      align="left"
    >
      Hover me
    </Tooltip>
  );
  expect(container).toMatchSnapshot();
});

it('renders Tooltip aligned top left unchanged', () => {
  const { container } = render(
    <Tooltip
      content="Here is some tooltip content by default as I am controlled"
      visible
      align="topLeft"
    >
      Hover me
    </Tooltip>
  );
  expect(container).toMatchSnapshot();
});

it('renders Tooltip aligned right unchanged', () => {
  const { container } = render(
    <Tooltip
      content="Here is some tooltip content by default as I am controlled"
      visible
      align="right"
    >
      Hover me
    </Tooltip>
  );
  expect(container).toMatchSnapshot();
});

it('renders Tooltip aligned top right unchanged', () => {
  const { container } = render(
    <Tooltip
      content="Here is some tooltip content by default as I am controlled"
      visible
      align="topRight"
    >
      Hover me
    </Tooltip>
  );
  expect(container).toMatchSnapshot();
});
