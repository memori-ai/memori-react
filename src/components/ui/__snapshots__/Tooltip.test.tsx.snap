// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tooltip aligned left unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-left memori-tooltip--visible"
  >
    <div
      class="memori-tooltip--content"
    >
      Here is some tooltip content by default as I am controlled
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;

exports[`renders Tooltip aligned right unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-right memori-tooltip--visible"
  >
    <div
      class="memori-tooltip--content"
    >
      Here is some tooltip content by default as I am controlled
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;

exports[`renders Tooltip aligned top left unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-topLeft memori-tooltip--visible"
  >
    <div
      class="memori-tooltip--content"
    >
      Here is some tooltip content by default as I am controlled
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;

exports[`renders Tooltip aligned top right unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-topRight memori-tooltip--visible"
  >
    <div
      class="memori-tooltip--content"
    >
      Here is some tooltip content by default as I am controlled
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;

exports[`renders Tooltip disabled unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-right memori-tooltip--disabled"
  >
    <div
      class="memori-tooltip--content"
    >
      Here is some tooltip content but I am disabled
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;

exports[`renders Tooltip unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-right"
  >
    <div
      class="memori-tooltip--content"
    >
      Here is some tooltip content
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;

exports[`renders Tooltip visible unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-right memori-tooltip--visible"
  >
    <div
      class="memori-tooltip--content"
    >
      Here is some tooltip content by default as I am controlled
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;

exports[`renders Tooltip with jsx content unchanged 1`] = `
<div>
  <div
    class="memori-tooltip memori-tooltip--align-right"
  >
    <div
      class="memori-tooltip--content"
    >
      <p>
        Here we have some JSX, useful for
        <a
          href="https://memori.ai"
          rel="noopener noreferrer"
          target="_blank"
        >
          links
        </a>
        , for example.
      </p>
    </div>
    <div
      class="memori-tooltip--trigger"
    >
      Hover me
    </div>
  </div>
</div>
`;
